local-volume/provisioner/pkg/util/volume_util.go

```diff
@@ -116,6 +135,23 @@ func (u *volumeUtil) GetFsAvailableByte(fullPath string) (uint64, error) {
 	return uint64(s.Frsize) * (s.Blocks - s.Bfree + s.Bavail), nil
 }
 
+// GetBlockSpaceByte returns  capacity in bytes of a block device.
+// fullPath is the pathname of block device.
+func (u *volumeUtil) GetBlockSpaceByte(fullPath string) (uint64, error) {
+	file, err := os.OpenFile(fullPath, os.O_RDWR, 0)
+	if err != nil {
+		return 0, err
+	}
+	defer file.Close()
+
+	var size int64
+	if _, _, err := unix.Syscall(unix.SYS_IOCTL, file.Fd(), unix.BLKGETSIZE64, uintptr(unsafe.Pointer(&size))); err != 0 {
```

Comment:
It fetches the size of block device in bytes into a 64 bit value (see http://www.microhowto.info/howto/get_the_size_of_a_linux_block_special_device_in_c.html).